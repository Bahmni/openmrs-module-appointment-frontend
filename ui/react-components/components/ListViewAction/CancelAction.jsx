import CustomPopup from "../CustomPopup/CustomPopup";
import {popup} from "./ListViewAction.module.scss";
import React from "react";
import useListViewAction from "./ListViewAction";

export default function CancelAction() {
  const [show, handleClose] = useListViewAction();

  function content() {
    return (
      <>
        <p>Are you sure, you want to mark appointment as Cancelled?</p>

        <button onClick={() => handleClose()}>Yes</button>
        <button onClick={() => handleClose()}>No</button>

      </>
    );
  }

  {
    return show &&
      <CustomPopup open={true} popupContent={content()} closeOnDocumentClick
                   closeOnEscape style={popup}/>
  }
};
