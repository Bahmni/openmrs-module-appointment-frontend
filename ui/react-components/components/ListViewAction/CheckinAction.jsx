import CustomPopup from "../CustomPopup/CustomPopup";
import {checkinPopup} from "./ListViewAction.module.scss";
import useListViewAction from "./useListViewAction";
import React from "react";

export default function CheckinAction() {
  const {show, handleClose} = useListViewAction();

  const currentTime = () => {
    const today = new Date();
    return `${today.getHours()}:${today.getMinutes()}`
  };

  const content = () => {
    return (
      <>
        {show &&
        <>
          <p>Are you sure, you want to mark appointment as CheckedIn?</p>
          <p>
            <b>Check in time:</b>
            <sup>*</sup>
            <input type="time" defaultValue={currentTime()}/>
          </p>
          <p>
            <button onClick={() => handleClose()}>Yes</button>
            <button onClick={() => handleClose()}>No</button>
          </p>
        </>}
      </>);
  };
  {
    return show &&
      (
        <CustomPopup open={true} popupContent={content()} closeOnDocumentClick
                     closeOnEscape style={checkinPopup}>
        </CustomPopup>
      );
  }
};
