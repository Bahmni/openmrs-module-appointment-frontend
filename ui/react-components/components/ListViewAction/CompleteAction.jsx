import CustomPopup from "../CustomPopup/CustomPopup";
import {popup} from "./ListViewAction.module.scss";
import React from "react";
import useListViewAction from "./ListViewAction";

export default function CompleteAction() {
  const [show, handleClose] = useListViewAction();

  function content() {
    return (
      <>
        <p>Are you sure, you want to mark appointment as Completed?</p>
        <button onClick={() => console.log("Clicked Yes")}>Yes</button>
        <button onClick={() => handleClose()}>No</button>
      </>
    )
  }

  {
    return show &&
      <CustomPopup open={true} popupContent={content()} closeOnDocumentClick closeOnEscape style={popup}/>
  }
};
