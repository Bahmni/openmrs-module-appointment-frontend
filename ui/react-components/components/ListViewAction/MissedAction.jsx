import CustomPopup from "../CustomPopup/CustomPopup";
import {popup} from "./ListViewAction.module.scss";
import React from "react";
import useListViewAction from "./ListViewAction";

export default function MissedAction() {
  const [show, handleClose] = useListViewAction();

  function content() {
    return (
      <>
        <p>Are you sure, you want to mark appointment as Missed?</p>

        <button onClick={() => handleClose()}>Yes</button>
        <button onClick={() => handleClose()}>No</button>

      </>
    );
  }

  {
    return show &&
      (
        <CustomPopup open={true} popupContent={content()} closeOnDocumentClick
                     closeOnEscape style={popup}>
        </CustomPopup>
      )
  }
};
