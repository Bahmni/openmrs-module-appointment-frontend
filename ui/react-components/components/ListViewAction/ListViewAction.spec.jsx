import React from "react";
import {render} from "@testing-library/react";
import CancelAction from "../ListViewAction/CancelAction";
import MissedAction from "../ListViewAction/MissedAction";
import CheckinAction from "../ListViewAction/CheckinAction";
import CompleteAction from "../ListViewAction/CompleteAction";

describe('List View Actions', () => {

  it('should render Cancel Action', () => {
    const {asFragment} = render(<CancelAction/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Complete Action', () => {
    const {asFragment} = render(<CompleteAction/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Missed Action', () => {
    const {asFragment} = render(<MissedAction/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Checkin Action', () => {
    const {asFragment} = render(<CheckinAction/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
