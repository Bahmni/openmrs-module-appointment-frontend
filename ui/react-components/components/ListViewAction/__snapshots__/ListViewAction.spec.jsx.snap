// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List View Actions should render Cancel Action 1`] = `
<DocumentFragment>
  <div
    style="position: absolute; top: 0px; left: 0px;"
  />
  <div
    class="popup-overlay popup popup-overlay-overlay"
    style="position: fixed; top: 0px; bottom: 0px; left: 0px; right: 0px; background: rgba(0, 0, 0, 0.5); display: flex; z-index: 999;"
  >
    <div
      class="popup-content popup popup-overlay-content"
      style="position: relative; background: rgb(255, 255, 255); width: 50%; margin: auto; border: 1px solid rgb(187, 187, 187); padding: 5px;"
    >
      <p>
        Are you sure, you want to mark appointment as Cancelled?
      </p>
      <button>
        Yes
      </button>
      <button>
        No
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`List View Actions should render Checkin Action 1`] = `
<DocumentFragment>
  <div
    style="position: absolute; top: 0px; left: 0px;"
  />
  <div
    class="popup-overlay checkinPopup popup-overlay-overlay"
    style="position: fixed; top: 0px; bottom: 0px; left: 0px; right: 0px; background: rgba(0, 0, 0, 0.5); display: flex; z-index: 999;"
  >
    <div
      class="popup-content checkinPopup popup-overlay-content"
      style="position: relative; background: rgb(255, 255, 255); width: 50%; margin: auto; border: 1px solid rgb(187, 187, 187); padding: 5px;"
    >
      <p>
        Are you sure, you want to mark appointment as CheckedIn?
      </p>
      <p>
        <b>
          Check in time:
        </b>
        <sup>
          *
        </sup>
        <input
          type="time"
          value="21:27"
        />
      </p>
      <p>
        <button>
          Yes
        </button>
        <button>
          No
        </button>
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`List View Actions should render Complete Action 1`] = `
<DocumentFragment>
  <div
    style="position: absolute; top: 0px; left: 0px;"
  />
  <div
    class="popup-overlay popup popup-overlay-overlay"
    style="position: fixed; top: 0px; bottom: 0px; left: 0px; right: 0px; background: rgba(0, 0, 0, 0.5); display: flex; z-index: 999;"
  >
    <div
      class="popup-content popup popup-overlay-content"
      style="position: relative; background: rgb(255, 255, 255); width: 50%; margin: auto; border: 1px solid rgb(187, 187, 187); padding: 5px;"
    >
      <p>
        Are you sure, you want to mark appointment as Completed?
      </p>
      <button>
        Yes
      </button>
      <button>
        No
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`List View Actions should render Missed Action 1`] = `
<DocumentFragment>
  <div
    style="position: absolute; top: 0px; left: 0px;"
  />
  <div
    class="popup-overlay popup popup-overlay-overlay"
    style="position: fixed; top: 0px; bottom: 0px; left: 0px; right: 0px; background: rgba(0, 0, 0, 0.5); display: flex; z-index: 999;"
  >
    <div
      class="popup-content popup popup-overlay-content"
      style="position: relative; background: rgb(255, 255, 255); width: 50%; margin: auto; border: 1px solid rgb(187, 187, 187); padding: 5px;"
    >
      <p>
        Are you sure, you want to mark appointment as Missed?
      </p>
      <button>
        Yes
      </button>
      <button>
        No
      </button>
    </div>
  </div>
</DocumentFragment>
`;
